torch/csrc/jit/tensorexpr/llvm_codegen.cpp

```diff
@@ -826,7 +827,21 @@ void LLVMCodeGenImpl::visit(const Cast* v) {
       } else {
         value_ = irb_.CreateSIToFP(value_, dstType);
       }
+
     } else if (dstType->isIntOrIntVectorTy()) {
+      // Bool is represented as 8bits, we need to make sure
+      // we're appropriately setting the one bit that is used
+      if (v->dtype().scalar_type() == ScalarType::Bool) {
+        llvm::Constant* constant_zero = llvm::ConstantInt::get(srcType, 0);
+        if (v->src_value()->dtype().lanes() != 1) {
```

Comment:
Do we have test coverage for the multi-lane case? If not, we should add it.

Reply:
The special case for boolean should be moved to a helper function. We have too many levels of nesting here.

Reply:
yes we do for multi-lane case. agreed i can reduce nesting.